windrose/windrose_main.py

```diff
@@ -0,0 +1,123 @@
+            file_list = os.listdir(site)
+            filtered_files = [
+                file for file in file_list
+                if "_BASE_" in file and "_HH_" in file
+            ]
+            # print(site)
+
+            for filtered_file in filtered_files:
+                file_path = site / filtered_file
+                df = pd.read_csv(file_path, skiprows=[0, 1])
+
+                if 'WS' not in df.columns or 'WD' not in df.columns:
+                    print("Both 'WS' and 'WD' or either columns are missing.")
+                else:
+                    self.process_season_data(df, site)
+
+    def process_season_data(self, df, site):
+        df_1 = df[(df['WD'] != -9999) | (df['WS'] != -9999)]
+        speed = df_1['WS']
+        direction = df_1['WD']
+        timestamp = df_1['TIMESTAMP_START']
+        time_series = pd.to_datetime(timestamp, format='%Y%m%d%H%M')
+        hourz = time_series.dt.strftime('%H%M').astype(int)
+        monthz = time_series.dt.month
+        year_start = time_series.dt.year.iloc[0]
+        year_end = time_series.dt.year.iloc[-1]
+
+        all_season_df = pd.DataFrame({
+            'month': monthz,
+            'hour': hourz,
+            'speed': speed,
+            'direction': direction
+        })
+
+        all_season = list(range(1, 13))
+        spring = [3, 4, 5]
+        summer = [6, 7, 8]
+        autumn = [9, 10, 11]
+        winter = [12, 1, 2]
+        day_time = list(range(600, 1810, 10))
+        night_time = list(range(0, 610, 10)) + list(range(1800, 2410, 10))
+        seasons = [all_season, spring, summer, autumn, winter]
+        day_night = [day_time, night_time]
+        season_names = ['Full_Year', 'Spring', 'Summer', 'Autumn', 'Winter']
+        time_names = ['Daytime', 'Nighttime']
+        ndirections = 16
+        bins1 = [0, 1, 4, 9, 16, 26, 37, 50]
+        percentile95 = np.percentile(speed, 95)
+        bins2 = np.linspace(0, percentile95, 8)
+        vwinds_list = [bins1, bins2]
+
+        analyzer = WindroseAnalysis()
+        print(site.name)
+
+        site_id = f'{site.name[4:10]}'
+        version = f'{year_start}_{year_end}'
+        if "BASE" in site.name:
+            data_product = f'{site.name[11:15]}'
+        # else:
+        #     data_product = f'{site.name[11:15]}'
+
+        directory_path = self.directory / site_id / data_product / version
+
+        if not directory_path.exists():
+            directory_path.mkdir(parents=True)
+
+        for j, vwinds in enumerate(vwinds_list):
+            for c, season in enumerate(seasons):
+                season_df = all_season_df[all_season_df['month'].isin(season)]
+                if c == 0:
+                    filename = directory_path / f"{season_names[c]}{j + 1}"
+                    analyzer.process_season_data(ndirections,
+                                                 vwinds, season_df, filename)
+                else:
+                    for t, ampm in enumerate(day_night):
```

Comment:
What is `ampm`?

Reply:
ampm is a list of hours representing either daytime or nighttime, depending on which part of the loop is currently executing. Here, ampm is a shorthand reference to AM/PM, indicating that the list contains hours for either the AM (daytime) or PM (nighttime) period.